surveys/views.py

```diff
@@ -15,21 +16,82 @@
 
 from .models import SurveyFormEntry, SurveyChart
 from .forms import StudyCreateForm, StudyAreaCreateForm, SurveyCreateForm, \
-                   SurveyChartForm
+                   SurveyChartForm, LocationCreateForm, LocationAreaCreateForm, \
+                   LocationLineCreateForm, AgencyCreateForm
 
 
 class AgencyCreate(CreateView):
+    form_class = AgencyCreateForm
     model = Agency
     template_name = "agency_create.html"
-    fields = '__all__'
     success_url = '/'
 
 
 class LocationCreate(CreateView):
-    model = Location
+    form_class = LocationCreateForm
+    form_class_location_area = LocationAreaCreateForm
+    form_class_location_line = LocationLineCreateForm
+
     template_name = "location_create.html"
-    fields = '__all__'
-    success_url = '/'
+    success_url = reverse_lazy('surveys-create')
+
+    def get_context_data(self, **kwargs):
+        context = super(LocationCreate, self).get_context_data(**kwargs)
+
+        context['form_location'] = self.form_class(prefix="location")
+        context['form_location_area'] = self.form_class_location_area(prefix="location-area")
+        context['form_location_line'] = self.form_class_location_line(prefix="location-line")
+
+        return context
+
+    def post(self, request, **kwargs):
+        form_location = LocationCreateForm(request.POST, prefix="location")
+        form_location_area = LocationAreaCreateForm(request.POST, prefix="location-area")
+        form_location_line = LocationLineCreateForm(request.POST, prefix="location-line")
+
+        print(form_location.errors)
+        print(form_location_area.errors)
+        print(form_location_line.errors)
```

Comment:
I found these helpful to debugging and suspect they'll be helpful again in the future. I've moved them to the bottom of this class so that they'll only print if a form is found invalid. Do you think they're reasonable to leave there?

Reply:
It's probably not harmful, but is there a reason that the validation errors aren't displaying in the forms UI?

Reply:
This might be a bigger conversation: right now the form uses client-side validation to make sure that all required fields are filled in properly, so displaying the errors in addition felt redundant to me. There are also hidden forms (ie, the LocationLine form when you're creating a LocationArea) that I didn't want to risk showing confusing errors for. I think Regina and I had flagged this as an approach to reconsider further into the project and that was lost in the shuffle. Maybe we could chat about this more outside of this PR? Curious to hear your thoughts validation best practices.

Reply:
We chatted about this a bit on Semaphor and decided to push off this work to #137.